### ISSUE TITLE
在CPU上的det模型训练问题

### ISSUE THREAD

**Issue by beiweixiaoxu** [PaddlePaddle/PaddleOCR#1058]
python tools/train.py -c configs/det/det_r50_vd_db.yml -o Global.pretrain_weights=./pretrain_models/ResNet50_vd_ssld_pretrained/
2020-10-29 17:29:26,296-INFO: {'Global': {'debug': False, 'algorithm': 'DB', 'use_gpu': False, 'epoch_num': 1200, 'log_smooth_window': 20, 'print_batch_step': 2, 'save_model_dir': './output/det_db/', 'save_epoch_step': 200, 'eval_batch_step': [5000, 5000], 'train_batch_size_per_card': 8, 'test_batch_size_per_card': 16, 'image_shape': [3, 640, 640], 'reader_yml': './configs/det/det_db_icdar15_reader.yml', 'pretrain_weights': './pretrain_models/ResNet50_vd_ssld_pretrained/', 'save_res_path': './output/det_db/predicts_db.txt', 'checkpoints': None, 'save_inference_dir': None}, 'Architecture': {'function': 'ppocr.modeling.architectures.det_model,DetModel'}, 'Backbone': {'function': 'ppocr.modeling.backbones.det_resnet_vd,ResNet', 'layers': 50}, 'Head': {'function': 'ppocr.modeling.heads.det_db_head,DBHead', 'model_name': 'large', 'k': 50, 'inner_channels': 256, 'out_channels': 2}, 'Loss': {'function': 'ppocr.modeling.losses.det_db_loss,DBLoss', 'balance_loss': True, 'main_loss_type': 'DiceLoss', 'alpha': 5, 'beta': 10, 'ohem_ratio': 3}, 'Optimizer': {'function': 'ppocr.optimizer,AdamDecay', 'base_lr': 0.001, 'beta1': 0.9, 'beta2': 0.999}, 'PostProcess': {'function': 'ppocr.postprocess.db_postprocess,DBPostProcess', 'thresh': 0.3, 'box_thresh': 0.7, 'max_candidates': 1000, 'unclip_ratio': 1.5}, 'TrainReader': {'reader_function': 'ppocr.data.det.dataset_traversal,TrainReader', 'process_function': 'ppocr.data.det.db_process,DBProcessTrain', 'num_workers': 8, 'img_set_dir': './train_data/icdar2015/text_localization/', 'label_file_path': './train_data/icdar2015/text_localization/train_icdar2015_label.txt'}, 'EvalReader': {'reader_function': 'ppocr.data.det.dataset_traversal,EvalTestReader', 'process_function': 'ppocr.data.det.db_process,DBProcessTest', 'img_set_dir': './train_data/icdar2015/text_localization/', 'label_file_path': './train_data/icdar2015/text_localization/test_icdar2015_label.txt', 'test_image_shape': [736, 1280]}, 'TestReader': {'reader_function': 'ppocr.data.det.dataset_traversal,EvalTestReader', 'process_function': 'ppocr.data.det.db_process,DBProcessTest', 'infer_img': None, 'img_set_dir': './train_data/icdar2015/text_localization/', 'label_file_path': './train_data/icdar2015/text_localization/test_icdar2015_label.txt', 'do_eval': True}}
preprocess done
3 640 640
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/heads/det_db_head.py:123
The behavior of expression A - B has been unified with elementwise_sub(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_sub(X, Y, axis=0) instead of A - B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:46
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:47
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:100
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:103
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:113
The behavior of expression A - B has been unified with elementwise_sub(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_sub(X, Y, axis=0) instead of A - B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
/home/ywzhang/anaconda3/envs/dachuang/lib/python3.7/site-packages/paddle/fluid/layers/math_op_patch.py:322: UserWarning: /home/ywzhang/xsy/PaddleOCR-develop/ppocr/modeling/losses/det_basic_loss.py:113
The behavior of expression A * B has been unified with elementwise_mul(X, Y, axis=-1) from Paddle 2.0. If your code works well in the older versions but crashes in this version, try to use elementwise_mul(X, Y, axis=0) instead of A * B. This transitional warning will be dropped in the future.
  op_type, op_type, EXPRESSION_MAP[method_name]))
2020-10-29 17:29:27,168-INFO: If regularizer of a Parameter has been set by 'fluid.ParamAttr' or 'fluid.WeightNormParamAttr' already. The Regularization[L2Decay, regularization_coeff=0.000000] in Optimizer will not take effect, and it will only be applied to other Parameters!
3 640 640
./train_data/icdar2015/text_localization/train_icdar2015_label.txt
import ujson error: No module named 'ujson' use json
2020-10-29 17:29:29,130-INFO: places would be ommited when DataLoader is not iterable
2020-10-29 17:29:30,437-INFO: Loading parameters from ./pretrain_models/ResNet50_vd_ssld_pretrained/...
2020-10-29 17:29:30,812-INFO: Finish initing model from ./pretrain_models/ResNet50_vd_ssld_pretrained/
2020-10-29 17:29:30,812-INFO: During the training process, after the 5000th iteration, an evaluation is run every 5000 iterations
W1029 17:34:59.718602  5703 fuse_all_reduce_op_pass.cc:75] Find all_reduce operators: 189. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 185.
Killed
出现这一行"W1029 17:34:59.718602  5703 fuse_all_reduce_op_pass.cc:75] Find all_reduce operators: 189. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 185."，之后，等了很漫长的一段时间，然后被kill了...请问我这个是正常的训练现象吗？

**Comment by beiweixiaoxu** [comment]
哦对了，训练之前，我是export CPU_NUM=14过的，因为我看服务器上的cpu cores是14个，感觉应该不是这个问题。

**Comment by LDOUBLEV** [comment]
你在CPU上训练resnet50 backbone的DB？你关注下百度的[AIstudio](https://aistudio.baidu.com/aistudio/index)，有免费的V100GPU可以用

**Comment by beiweixiaoxu** [comment]
> 你在CPU上训练resnet50 backbone的DB？你关注下百度的[AIstudio](https://aistudio.baidu.com/aistudio/index)，有免费的V100GPU可以用

哦。。哈哈 感谢回答
